sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3994,10 +3994,45 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         c.markAsAnalyzed(AnalysisContext.get)
       case c: KeepAnalyzedQuery if c.resolved =>
         c.storeAnalyzedQuery()
-      case p: LogicalPlan if p.resolved =>
```

Comment:
doesn't the previous code work for CreateTable/Replace already?

Reply:
+1, the idea was to be generic here. If we need special handling for other commands, they should caught earlier in the rule.

Reply:
yea I think this is from a previous version of Anton's patch, let me check and update this